src/kmers.nim

```diff
@@ -321,3 +341,15 @@ proc search*(target: pot_t; query: pot_t): deques.Deque[seed_pair_t] =
         inc(i)
 
     return hit_stack
+
+## This function counts the number of uniq kmers in the pot if searchable if not
+## the function calls make searchable.
+## @param pot_t - a ref to a pot_t
+## TODO: add test coverage
+#
+proc nuniq*(pot: pot_t): int =
+ if searchable(pot):
```

Comment:
```nim
make_searchable(pot)
return len(pot.ht)
```
Sure, there is still an if-statement inside `make_searchable()`, but we don't have an *extra* if-statement.

By the way, unit-tests should try to test all branches. That's another reason to avoid extra branches.